lib/anonymousNaming.js

```diff
@@ -43,22 +49,95 @@ function assignUidToParameterSchemas(doc) {
     assignIdToParameters(channel.parameters());
   });
 }
-  
+
 /**
  * Assign uid to component schemas. 
  * 
  * @private
  * @param {AsyncAPIDocument} doc 
+ * @param {boolean} traversIds Walk recursive through schema and try to find reasonamble uid where no uids was given.
  */
-function assignUidToComponentSchemas(doc) {
+function assignUidToComponentSchemas(doc, traversIds = false) {
   if (doc.hasComponents()) {
-    for (const [key, s] of Object.entries(doc.components().schemas())) {
-      s.json()[String(xParserSchemaId)] = key;
+    for (const [key, schema] of Object.entries(doc.components().schemas())) {
+      if (traversIds) {
+        assignUidToComponentSchemasRecursive(schema, key);
+      } else {
+        schema.json()[String(xParserSchemaId)] = key;
+      }
     }
   }
 }
 
-/**
+function assignUidToComponentSchemasRecursive(schema, key) {
```

Comment:
please add jsdoc like for other functions,

Reply:
done